components/Typography/examples/TypographyMaterialScalableStylesViewController.m

```diff
@@ -0,0 +1,173 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "TypographyMaterialScalableStylesViewController.h"
+
+#import "MaterialTypography.h"
+#import "MaterialTypographyScheme.h"
+
+@implementation TypographyMaterialScalableStyleViewController {
+  NSArray<NSString *> *_strings;
+  NSArray<NSString *> *_styleNames;
+  NSArray<UIFont *> *_styleFonts;
+  MDCTypographyScheme *_typography;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  self.tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
+
+  self.tableView.rowHeight = UITableViewAutomaticDimension;
+  self.tableView.estimatedRowHeight = 50.0;
+
+  _strings = @[
+    @"Material Design Components", @"A quick brown fox jumped over the lazy dog.",
+    @"ABCDEFGHIJKLMNOPQRSTUVWXYZ", @"abcdefghijklmnopqrstuvwxyz", @"1234567890",
+    @"!@#$%^&*()-=_+[]\\;',./<>?:\""
+  ];
+
+  _styleNames = @[
+    // Material Font Styles
+    @"Body1",     @"Body1 (Scalable)",     @"Body2",     @"Body2 (Scalable)",
+    @"Caption",   @"Caption (Scalable)",   @"Button",    @"Button (Scalable)",
+    @"Overline",  @"Overline (Scalable)",  @"Subtitle1", @"Subtitle1 (Scalable)",
+    @"Subtitle2", @"Subtitle2 (Scalable)", @"Headline1", @"Headline1 (Scalable)",
+    @"Headline2", @"Headline2 (Scalable)", @"Headline3", @"Headline3 (Scalable)",
+    @"Headline4", @"Headline4 (Scalable)", @"Headline5", @"Headline5 (Scalable)",
+    @"Headline6", @"Headline6 (Scalable)",
+  ];
+
+  _typography =
+      [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201902];
+
+  _styleFonts = @[
```

Comment:
Please merge `styleFonts` and `styleNames`  into a dictionary so the pairing isn't so brittle.

Reply:
I'm not sure I follow.  One contains fonts, the other strings.  Are you recommending a tuple?  I don't think a custom object is warranted here.
Documentation added.

Reply:
I'm suggesting `NSDictionary<NSString *, UIFont *> *` as the type.

Reply:
Ah, now I follow.  Something to note is that _styleFonts actually changes, _styleNames does not.
For context, this code comes from TypographyMaterialStylesViewController.